code/API_definitions/qod-api.yaml

```diff
@@ -358,42 +356,75 @@ components:
         * `QOS_S` - Qualifier for the requested QoS profile _S_
         * `QOS_M` - Qualifier for the requested QoS profile _M_
         * `QOS_L` - Qualifier for the requested QoS profile _L_
-    
-    Notification:
+
+    EventNotification:
       type: object
       required:
-        - sessionId
-        - eventReports
+        - event
       properties:
-        sessionId:
-          $ref: "#/components/schemas/SessionId"
-        eventReports:
-          type: array
-          items:
-            $ref: "#/components/schemas/EventReport"
-          minItems: 1
-          description: Contains the reported event and applicable information
-    
-    EventReport:
-      oneOf:
-        - $ref: '#/components/schemas/QosStatusChangedEventReport'
+        event:
+          $ref: "#/components/schemas/Event"
+
+    Event:
+      anyOf:
+        - $ref: "#/components/schemas/QosStatusChangedEvent"
       discriminator:
         propertyName: eventType
         mapping:
-          QOS_STATUS_CHANGED: "#/components/schemas/QosStatusChangedEventReport"
-        
-    QosStatusChangedEventReport:
+          QOS_STATUS_CHANGED: "#/components/schemas/QosStatusChangedEvent"
+      description: Notification event
+
+    EventId:
+      type: string
+      format: uuid
+      example: 5698d710-9b1b-4695-a958-7b228f08128c
+      description: Unique identifier of the event
+
+    EventType:
+      type: string
+      enum:
+        - QOS_STATUS_CHANGED
+      description: Type of the event
+
+    EventTime:
+      type: string
+      format: date-time
+      example: 2023-05-30T10:18:28Z
+      description: Date time when the event occurred
+
+    EventBase:
       type: object
       required:
         - eventType
-        - qosStatus
+        - eventTime
       properties:
+        eventid:
+          $ref: "#/components/schemas/EventId"
         eventType:
-          $ref: "#/components/schemas/SessionEventType"
-        qosStatus:
-          $ref: "#/components/schemas/QosStatus"
-        statusInfo:
-          $ref: "#/components/schemas/StatusInfo"
+          $ref: "#/components/schemas/EventType"
+        eventTime:
+          $ref: "#/components/schemas/EventTime"
+
+    QosStatusChangedEvent:
+      allOf:
+        - $ref: "#/components/schemas/EventBase"
+        - type: object
+          properties:
+            eventDetail:
+              type: object
+              description: Event details depending on the event type
+              required:
+                - sessionId
+                - qosStatus
+              properties:
+                sessionId:
+                  $ref: "#/components/schemas/SessionId"
+                qosStatus:
+                  $ref: "#/components/schemas/QosStatus"
```

Comment:
@sfnuser yes I think so, but do you think we need a change in the api-spec for that?

Reply:
@akoshunyadi - Yes please. We may have to define another `object` that exposes only those two states and reference it in the `eventDetail` object. Considering we decided to have the spec as the main documentation, this change will make it clear a rollback to `requested` state is not allowed/possible.

Reply:
@sfnuser This spec only says that in case of the event QOS_STATUS_CHANGED the eventDetails contains the current qosStatus. Generally I think if a specific data type is referenced, it doesn't mean that in each use case all of the possible values are used.

Reply:
>Generally I think if a specific data type is referenced, it doesn't mean that in each use case all of the possible values are used.

@akoshunyadi I tend to disagree. How do we explain this to a tester writing a test case using the spec?

Reply:
@sfnuser We could add it as documentation to the spec, in [PR151](https://github.com/camaraproject/QualityOnDemand/pull/151)